libcontainer/dmz/_dmz.c

```diff
@@ -11,5 +14,14 @@ int main(int argc, char **argv)
 {
 	if (argc < 1)
 		return 127;
-	return execve(argv[0], argv, environ);
+	int ret = execve(argv[0], argv, environ);
+	if (ret) {
+		/* NOTE: This error message format MUST match Go's format. */
+		char err_msg[5 + PATH_MAX + 1] = "exec ";	// "exec " + argv[0] + '\0'
+		strncat(err_msg, argv[0], PATH_MAX);
```

Comment:
Ah, I didn't notice you switched to `strncat`. In that case, the length should really be `sizeof(err_msg)`.

Reply:
Why? The C library function char *strncat(char *dest, const char *src, size_t n) appends the string pointed to by src to the end of the string pointed to by dest up to n characters long. So I think it should be `PATH_MAX`.

Reply:
My bad, I was thinking of `strlcat` (which is what the kernel uses these days) where the length argument is the size of the buffer not the maximum number of bytes to copy.

Reply:
Sorry for the confusion, I mentioned the change to strncat here: https://github.com/opencontainers/runc/pull/4172#discussion_r1460922408